modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateProcessor.java

```diff
@@ -64,7 +64,7 @@
 
     @Override
     public void execute(IngestDocument ingestDocument) {
-        String value = ingestDocument.getFieldValue(field, String.class);
+        String value = Objects.toString(ingestDocument.getFieldValue(field, Object.class));
```

Comment:
do we really want to be lenient with nulls?

Reply:
@jpountz `IngestDocument#getFieldValue(...)` fails if there is no value for the specified field,  so I think the current code is good?

Reply:
Does it also fail if there is a value but this value is `null`?

Reply:
No, it doesn't fail here, but later at line 81, because the date can't be parsed. That is the same behaviour as before when null was specified, only now the `lastException` exception message is different. (`java.lang.NumberFormatException: For input string: "null"` instead of `java.lang.NumberFormatException: For input string: null`)

Reply:
@jpountz I've added an if check here and in `DateIndexNameProcessor` too.